vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp

```diff
@@ -87,28 +92,32 @@ void MapBufferBuilder::putNull(Key key) {
   putInt(key, NULL_VALUE);
 }
 
-void MapBufferBuilder::putInt(Key key, int value) {
+void MapBufferBuilder::putInt(Key key, int32_t value) {
   uint8_t *bytePointer = reinterpret_cast<uint8_t *>(&(value));
   storeKeyValue(key, bytePointer, INT_SIZE);
 }
 
-void MapBufferBuilder::ensureDynamicDataSpace(int size) {
+void MapBufferBuilder::ensureDynamicDataSpace(int32_t size) {
   if (dynamicDataValues_ == nullptr) {
```

Comment:
this code should probably assert `size >= 0`. Still don't see why it is not backed by a std: collections...

Reply:
I haven't payed much attention to what the data structure is used for, but std::unordered_map is infamous for being slow relative to more cache optimized hash tables.

The standard for unordered_map effectively dictates that is used chained hashing/linked lists. This is because it requires iterators to have pointer stability. This does a lot of pointer chasing, where as a buffer based structure (like open addressing) is much faster on modern hardware.